graphql_compiler/cost_estimation/analysis.py

```diff
@@ -0,0 +1,238 @@
+        - query_metadata: a representation of the query
+        - parameters: the parameters used for the query
+
+    Returns:
+        Dict mapping some PropertyPaths to the interval of filtered values at that field.
+    """
+    field_value_intervals = {}
+    for location, location_info in query_metadata.registered_locations:
+        filter_infos = query_metadata.get_filter_infos(location)
+        vertex_type_name = location_info.type.name
+
+        # Group filters by field
+        single_field_filters: Dict[str, List[FilterInfo]] = {}
+        for filter_info in filter_infos:
+            if len(filter_info.fields) == 0:
+                raise AssertionError(f"Got filter on 0 fields {filter_info} on {vertex_type_name}")
+            elif len(filter_info.fields) == 1:
+                single_field_filters.setdefault(filter_info.fields[0], []).append(filter_info)
+            else:
+                pass  # We don't do anything for multi-field filters yet
+
+        # Find field_value_interval for each field
+        for field_name, filters_on_field in single_field_filters.items():
+            filters_on_field = [
+                filter_info for filter_info in filter_infos if filter_info.fields == (field_name,)
+            ]
+
+            integer_interval = get_integer_interval_for_filters_on_field(
+                schema_info, filters_on_field, vertex_type_name, field_name, parameters
+            )
+            field_value_interval = _convert_int_interval_to_field_value_interval(
+                schema_info, vertex_type_name, field_name, integer_interval
+            )
+
+            field_value_intervals[(location.query_path, field_name)] = field_value_interval
+    return field_value_intervals
+
+
+def get_distinct_result_set_estimates(
+    schema_info: QueryPlanningSchemaInfo,
+    query_metadata: QueryMetadataTable,
+    parameters: Dict[str, Any],
+) -> Dict[VertexPath, float]:
+    """Map each VertexPath in the query to its distinct result set estimate.
+
+    The distinct result set estimate for a query node is the expected number of different
+    vertices that will appear under it in the result of the query.
+
+    Args:
+        - schema_info: QueryPlanningSchemaInfo
+        - query_metadata: a representation of the query
+        - parameters: the query parameters
+
+    Returns:
+        the distinct result set estimate for each VertexPath
+    """
+    distinct_result_set_estimates = {}
+    for location, location_info in query_metadata.registered_locations:
+        vertex_type_name = location_info.type.name
+        filter_infos = query_metadata.get_filter_infos(location)
+        class_count = schema_info.statistics.get_class_count(vertex_type_name)
+        distinct_result_set_estimates[location.query_path] = adjust_counts_for_filters(
+            schema_info, filter_infos, parameters, vertex_type_name, class_count
+        )
+
+    # TODO transfer along many-to-one edges
```

Comment:
Could you please make this TODO a bit more clear? Not sure I understand what it means. Thanks!